extensions/amp-mathml/1.0/storybook/Basic.amp.js

```diff
@@ -0,0 +1,49 @@
+import {withAmp} from '@ampproject/storybook-addon';
+
+import * as Preact from '#preact';
+
+const formulas = {
+  quadraticFormula: '\\[x = {-b \\pm \\sqrt{b^2-4ac} \\over 2a}.\\]',
+  cauchyFormula: '\\[f(a) = \\frac{1}{2\\pi i} \\oint\\frac{f(z)}{z-a}dz\\]',
+  doubleAngleFormula: '$$ \\cos(θ+φ)=\\cos(θ)\\cos(φ)−\\sin(θ)\\sin(φ) $$',
+};
+
+export default {
+  title: 'amp-mathml-1_0',
+  decorators: [withAmp],
+  parameters: {
+    extensions: [{name: 'amp-mathml', version: '1.0'}],
+    experiments: ['bento'],
+  },
+  args: {
+    formula: formulas.quadraticFormula,
+  },
+};
+
+// DO NOT SUBMIT: This is example code only.
+export const _default = (formula, args) => {
+  return (
+    <>
+      <h2>The Quadratic Formula</h2>
+      <amp-mathml data-formula={formula} {...args}></amp-mathml>
```

Comment:
```suggestion
     <amp-mathml
        height="40"
        data-formula={formula.formula}
        {...args}
      ></amp-mathml>
```

Reply:
> 1. AMP layer is broken - it doesn't seem to pass through props correctly to the Preact component. I've declared the props in the `base-element.js` file, but not sure if there's anything extra I need to do. The Preact component seems to behave as expected. See AMP/Preact storybooks for repro.

Noticed two things here:
(1) The AMP layer needs defined dimensions to render. 

The legacy/classic version of the component supports `layout=container`, which causes content shifts due to when the formula iframe is inserted into the otherwise empty `<amp-mathml>` container. We should change this in 1.0 to improve performance, specifically to require at least a height dimension. 

The `ContainWrapper` at the root of the `ProxyIframeEmbed` tree of helper fns is our tool for enforcing defined dimensions before components are renderable. I'm seeing that for the given formulas, `40px` is a sufficient default. When I add `height="40"` attr to the `<amp-mathml>`, it should be able to display correctly.

(2) The AMP-mode Storybook appears to pass `formula` as an object, not a str. I notice that setting `data-formula={formula.formula}` solves the trick, though I'm not sure why.

Reply:
Re: (1) So what's the recommended solution here? Require that users set explicit heights? Or set 40px as a default height until the formula resizes?

Re: (2): This was my mistake - Storybook passes in an object of arguments, of which formula is one. That should've been `({formula, ...args})` not `(formula, ...args)`. Didn't realize this because nothing rendered even when I set it to a literal string because of (1)

Reply:
Re: (1) Actually, I'm not sure if `40px` will always work - for example, if the formula is inlined - right?

Reply:
> So what's the recommended solution here? Require that users set explicit heights? Or set 40px as a default height until the formula resizes?

It depends. I think 40px is a reasonable but potentially imperfect default height. We should test what happens if the formula is a super wide or tall one -- does it try to line wrap somehow for narrow screens? If its natural dimensions are bigger than 40, does it resize proportionally to 40 or overflow within its container? If any of these happens, and we can't correct it with `100%` width and height on the iframe, then we likely need to require users set dimensions intentionally instead to account for these cases manually.

Reply:
resolved offline - decided to require consumers to set dimensions explicitly because the dimensions are totally dependent on the formula